### ISSUE TITLE
Focusing on window doesn't allow editing document

### ISSUE THREAD

**Issue by hbsmith** [typora/typora-issues#5921]
- [x ] Searched existing issues to avoid creating duplicates.

- [x ] Confirmed that it can be reproduced in built-in themes without customized css.

   If it only exists in 3rd party themes or css, you can still report it, but please attach the theme target or the css file. We may not "fix" it, if it is caused by 3rd party themes or css styles that we cannot support.

- [x ] Searched http://support.typora.io/

----

**Describe the bug**
In large files, when focused on another window (e.g. chrome), and then clicking back into the typora window, the typora window focuses, but typing does not show up in the document. Seems to 

**To Reproduce**
1. Open a large typora document.
2. Click on another window
3. Click back into the typora window once (the typora window will be focused) 
4. Start typing (error manifests that nothing will appear/change in the typora document)

**Expected behavior**
I expect that as soon as I focus on the typora window, typing will show up in the document.

**Screenshots / Screencasts**
If applicable, add screenshots / screencasts to help explain your problem.

**Sample Markdown File**
The error occurs consistently for me in my file with ~23k words. (143KB)

**Desktop (please complete the following information):**
 - OS: MacOS Sonoma 14.1.2

**Typora Version**
e.g: 1.7.6

**Additional context**
Might be related to #5880 , as I have noticed a LOT of performance issues since upgrading to MacOS Sonoma. Large files are almost unusable which is a problem because all of my notes files are long typora files.

**Comment by FlorentinGuth** [comment]
I also have this issue (MacOs 14.5, Typora  1.9.3).
~I noticed that switching windows with Cmd-Tab does not have this issue, only focusing Typora with a mouse click leads to the "focused but unable to type" state.~
EDIT: not sure about that anymore, I've been unable to find something that consistently reproduces or avoids the bug.

**Comment by abnerlee** [comment]
Is there a caret when focus to Typora window?

**Comment by FlorentinGuth** [comment]
Yes, there is a caret, the window "looks" focused in all ways, except that it seems to ignore most keyboard inputs (I can save with Cmd-S, but cannot move the cursor nor type text).
I've edited my above comment: I've tried finding steps that consistently reproduce the bug (or avoid it) but I've not been able to see which parameters matter. Clicking on another window then clicking on Typora's window again seems to be the  most consistent way but it does not always happen.

**Comment by michaelbaisch** [comment]
This is still happening for me; already tested `1.14.3-dev` and I am on `macOS 26.5.1`.

https://github.com/user-attachments/assets/e5e4cf64-e9b1-4a09-b8f9-7be59fdc4093

**Comment by hbsmith** [comment]
Would be great for this to be fixed. At least once per week I start typing while in a meeting and then look and realize that even though i clicked on the window the text isn't getting input. IMO this is significant undesired behavior for a text editor.
